alibi/explainers/tests/test_kernel_shap.py

```diff
@@ -310,12 +325,35 @@ def test_sum_categories(n_feats, feat_enc_dim, start_idx):
     # check that if inputs are correct, we retrieve the sum in the correct col
     else:
         summ_X = sum_categories(X, start_idx, feat_enc_dim)
-        assert summ_X.shape[1] == X.shape[1] - sum(feat_enc_dim) + len(feat_enc_dim)
+        assert summ_X.shape[-1] == X.shape[-1] - sum(feat_enc_dim) + len(feat_enc_dim)
+        if ndim == 3:
+            assert summ_X.shape[-2] == X.shape[-2] - sum(feat_enc_dim) + len(feat_enc_dim)
```

Comment:
Slightly unclear what the last dimension in the 3d case is for. I assume the assumption here is that the 2nd dimension (although why not use the index `1` for both cases for clarity?) are the feature encodings.

Reply:
I added detailed documentation which should clear any confusion. Also changed the assertion style to reflect better what the docs say.